tensorrt_llm/_torch/models/modeling_qwen3vl.py

```diff
@@ -0,0 +1,843 @@
+                torch.cat(pixel_values_list, dim=0)
+                if len(pixel_values_list) > 1
+                else pixel_values_list[0]
+            )
+        if pixel_values_videos_list:
+            mm_content_dict["pixel_values_videos"] = (
+                torch.cat(pixel_values_videos_list, dim=0)
+                if len(pixel_values_videos_list) > 1
+                else pixel_values_videos_list[0]
+            )
+
+        # Prepare extra data
+        mm_extra_data = {}
+        if image_grid_thw_list:
+            mm_extra_data["image_grid_thw"] = (
+                torch.cat(image_grid_thw_list, dim=0)
+                if len(image_grid_thw_list) > 1
+                else image_grid_thw_list[0]
+            )
+        if video_grid_thw_list:
+            mm_extra_data["video_grid_thw"] = (
+                torch.cat(video_grid_thw_list, dim=0)
+                if len(video_grid_thw_list) > 1
+                else video_grid_thw_list[0]
+            )
+
+        return mm_content_dict, mm_extra_data
+
+    @torch.inference_mode()
+    def forward(self, multimodal_params: List[MultimodalParams]):
+        mm_content_data, mm_extra_data = self._parse_and_batch_multimodal_data(multimodal_params)
+        pixel_values = mm_content_data.get("pixel_values", None)
+        pixel_values_videos = mm_content_data.get("pixel_values_videos", None)
+
+        image_grid_thw = mm_extra_data.get("image_grid_thw", None)
+        video_grid_thw = mm_extra_data.get("video_grid_thw", None)
+
+        embeds = []
+        deepstack_embeds = []
+        if pixel_values is not None:
+            pixel_values = pixel_values.to(self.model_dtype)
+            image_embeds, deepstack_image_embeds = self.visual(
+                pixel_values, grid_thw=image_grid_thw
+            )
+            # print("shapes", image_embeds.shape, len(deepstack_image_embeds))
+            embeds.append(image_embeds)
+            deepstack_embeds.append(torch.stack(deepstack_image_embeds))
+
+        if pixel_values_videos is not None:
+            pixel_values_videos = pixel_values_videos.to(self.model_dtype)
+            video_embeds, deepstack_video_embeds = self.visual(
+                pixel_values_videos, grid_thw=video_grid_thw
+            )
+            embeds.append(video_embeds)
+            deepstack_embeds.append(torch.stack(deepstack_video_embeds))
+        return embeds, deepstack_embeds
+
+
+class Qwen3VLModelBase(PreTrainedModel):
+    def __init__(
+        self,
+        model_config: ModelConfig[PretrainedConfig],
+        *args,
+        **kwargs,
+    ) -> None:
+        model_config.pretrained_config.text_config.rope_scaling["type"] = "mrope"
+        self.original_arch = model_config.pretrained_config.architectures[0]
+        config = model_config.pretrained_config
+
+        assert model_config.attn_backend == "TRTLLM", "Qwen3-VL only supports TRTLLM backend now"
+        super().__init__(config)
+        self.init_mrope_embedding(model_config)
```

Comment:
like in qwen2.5-vl, do we need `disable_fuse_rope` option here?

Also, a quick question, maybe to @yechank-nvidia , if we enable `disable_fuse_rope` (so cos_sin not computed in vit), where are the rope_cos_sin values computed?